src/python/pants/core/goals/publish_test.py

```diff
@@ -0,0 +1,172 @@
+    required_fields = (MockRepositoriesField,)
+    publish_request_type = MockPublish
+
+    repositories: MockRepositoriesField
+
+
+@rule
+async def mock_publish(request: MockPublish) -> PublishPackagesProcesses:
+    if not request.field_set.repositories.value:
+        return PublishPackagesProcesses(())
+
+    return PublishPackagesProcesses(
+        tuple(
+            PublishPackageProcesses(
+                names=tuple(
+                    artifact.relpath
+                    for pkg in request.packages
+                    for artifact in pkg.artifacts
+                    if artifact.relpath
+                ),
+                processes=() if repo == "skip" else (InteractiveProcess(["echo", repo]),),
+            )
+            for repo in request.field_set.repositories.value
+        )
+    )
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    return RuleRunner(
+        rules=[
+            *core_rules(),
+            *python_rules(),
+            *rules(),
+            mock_publish,
+            PythonDistribution.register_plugin_field(MockRepositoriesField),
+            UnionRule(PublishFieldSet, PublishTestFieldSet),
+            UnionRule(PublishPackagesRequest, MockPublish),
+        ],
+        target_types=[PythonLibrary, PythonDistribution],
+        objects={"python_artifact": PythonArtifact},
+    )
+
+
+def test_noop(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files(
+        {
+            "src/BUILD": dedent(
+                """\
+                python_library()
+                python_distribution(
+                  name="dist",
+                  provides=python_artifact(
+                    name="my-package",
+                    version="0.1.0",
+                  ),
+                )
+                """
+            ),
+        }
+    )
+
+    result = rule_runner.run_goal_rule(
+        Publish,
+        args=("src:dist",),
+        env_inherit={"HOME", "PATH", "PYENV_ROOT"},
+    )
+
+    assert result.exit_code == 0
+    assert "Nothing published." in result.stderr
+
+
+def test_skipped_publish(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files(
+        {
+            "src/BUILD": dedent(
+                """\
+                python_library()
+                python_distribution(
+                  name="dist",
+                  provides=python_artifact(
+                    name="my-package",
+                    version="0.1.0",
+                  ),
+                  repositories=["skip"],
+                )
+                """
+            ),
+        }
+    )
+
+    result = rule_runner.run_goal_rule(
+        Publish,
+        args=("src:dist",),
+        env_inherit={"HOME", "PATH", "PYENV_ROOT"},
+    )
+
+    assert result.exit_code == 0
+    assert "my-package-0.1.0.tar.gz skipped." in result.stderr
+    assert "my_package-0.1.0-py3-none-any.whl skipped." in result.stderr
+
+
+@pytest.mark.skip("Can not run interactive process from test..?")
```

Comment:
As the comment says, is there a way to run a goal rule using interactive runner somehow?